src/app/core/session/login/login.component.ts

```diff
@@ -45,64 +44,61 @@ export class LoginComponent {
 
   constructor(
     private _sessionService: SessionService,
-    @Optional() private router: Router,
-    @Optional() private route: ActivatedRoute
+    private loggingService: LoggingService,
+    private router: Router,
+    private route: ActivatedRoute
   ) {}
 
   /**
    * Do a login with the SessionService.
    */
   login() {
     this.loginInProgress = true;
+    this.errorMessage = "";
 
     this._sessionService
       .login(this.username, this.password)
       .then((loginState) => {
-        if (loginState === LoginState.LOGGED_IN) {
-          this.onLoginSuccess();
-        } else {
-          if (
-            this._sessionService.getSyncState().getState() ===
-              SyncState.ABORTED &&
-            this._sessionService.getConnectionState().getState() ===
-              ConnectionState.OFFLINE
-          ) {
+        switch (loginState) {
+          case LoginState.LOGGED_IN:
+            this.onLoginSuccess();
+            break;
+          case LoginState.UNAVAILABLE:
             this.onLoginFailure(
-              $localize`Can't login for the first time when offline. Please try again later.`
+              $localize`:LoginError:Please connect to the internet and try again`
             );
-          } else if (
-            this._sessionService.getConnectionState().getState() ===
-            ConnectionState.OFFLINE
-          ) {
+            break;
+          case LoginState.LOGIN_FAILED:
             this.onLoginFailure(
-              $localize`Username or password incorrect!
-               You might also face this problem because you are currently offline.
-               Please connect to the internet to synchronize the latest user data.`
+              $localize`:LoginError:Username and/or password incorrect`
+            );
+            break;
+          default:
+            throw new Error(
+              $localize`:LoginError:Unexpected login state: ${loginState}`
             );
```

Comment:
why localize the error object when it is caught and a more user-friendly message is set below?